### PR TITLE
[TR] Merge 1.4 to master

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ before_install:
     - if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then
         phpenv config-rm xdebug.ini;
       fi
-    - echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
+    - echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
```
Comment (pierallard) [inline .travis.yml]: To remove

File: features/Context/WebUser.php [hunk 2]
```diff
@@ -2736,4 +2750,27 @@ public function theApiKeyShouldBe($not, $value)
...
+     *
+     * @Then /^I should see (\d+) items? in the autocomplete$/
+     */
+    public function iShouldSeeAutocompleteItems($expectedCount)
+    {
+        $items = $this->spin(function () {
+            return $this
+                ->getCurrentPage()
+                ->findAll('css', '.select2-results .select2-result-selectable, .select2-results .select2-no-results');
+        }, 20, 'Cannot find any select2 items');
```
Comment (skeleton) [inline features/Context/WebUser.php]: remove `20` param

File: spec/Pim/Bundle/CatalogBundle/Validator/Constraints/UniqueVariantAxisValidatorSpec.php [hunk 3]
```diff
@@ -270,3 +269,8 @@ function it_adds_a_violation_when_validating_product_in_groups_with_non_unique_c
         $this->validate($redTShirtProduct, $uniqueVariantAxisConstraint);
     }
 }
+
+interface ProductRepositoryInterface extends BaseProductRepositoryInterface
```
Comment (skeleton) [inline spec/Pim/Bundle/CatalogBundle/Validator/Constraints/UniqueVariantAxisValidatorSpec.php]: remove it (findProductIdsForVariantGroup method has to be added to ProductRepositoryInterface)

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Repository/ProductRepository.php [hunk 4]
```diff
@@ -729,4 +726,55 @@ public function countAll()
...
         return $count;
     }
+
+    /**
+     * @param GroupInterface $variantGroup
+     * @param array          $criteria
+     *
+     * @return array
+     */
+    public function findProductIdsForVariantGroup(GroupInterface $variantGroup, array $criteria = [])
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Repository/ProductRepository.php]: add it to interface and as it's a bc break, don't forget to add it in changelog

File: features/Context/DataGridContext.php [hunk 5]
```diff
@@ -855,11 +855,12 @@ public function iRefrestTheGrid()
      */
     public function iClickBackToGrid()
     {
-        $this->spin(function () {
-            $this->getSession()->getPage()->clickLink('Back to grid');
-
-            return true;
+        $this->wait();
```
Comment (phaseinducer) [inline features/Context/DataGridContext.php]: why ? :'(

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/attribute-fetcher.js [hunk 6]
```diff
@@ -36,12 +36,23 @@ define(['jquery', 'underscore', 'pim/base-fetcher', 'routing'], function ($, _,
...
         },
 
         /**
+         * This method overrides the base method, to send a POST query instead of a GET query, because the request
+         * URI can be too long.
+         * TODO Should be deleted to set it back to GET.
+         *
+         * {@inheritdoc}
+         */
+        getJSON: function (url, parameters) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/attribute-fetcher.js]: We should rework this

File: features/Context/DataGridContext.php (hunk 5 above)
Comment (pierallard) [inline features/Context/DataGridContext.php]: Is this the "wait" it's disturbing you or the 2nd version of this method ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/attribute-fetcher.js (hunk 6 above)
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/attribute-fetcher.js]: @phaseinducer There is a card in the board for doing this in a next sprint

File: features/Context/DataGridContext.php (hunk 5 above)
Comment (phaseinducer) [inline features/Context/DataGridContext.php]: The wait. We try to remove them, plus we should never put a wait at the beginning of a step.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/attribute-fetcher.js (hunk 6 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/attribute-fetcher.js]: Ok perfect

File: behat.yml.dist [hunk 7]
```diff
@@ -33,7 +33,7 @@ jenkins-1:
             kernel:
                 env: behat1
     formatter:
-        name: Context\SummaryFormatter, Context\Formatter\JUnitFormatter
+        name: Context\SummaryFormatter,junit
```
Comment (juliensnz) [inline behat.yml.dist]: Do we need that ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/attribute-fetcher.js (hunk 6 above)
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/attribute-fetcher.js]: :+1:

File: src/Pim/Bundle/PdfGeneratorBundle/Twig/ImageExtension.php [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+ */
+class ImageExtension extends \Twig_Extension
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return [
+            'image_path' => new \Twig_Function_Method($this, 'getImagePath'),
```
Comment (juliensnz) [inline src/Pim/Bundle/PdfGeneratorBundle/Twig/ImageExtension.php]: maybe a more specialized name

**Comment by juliensnz** [comment]
GTM after comment fixes

File: features/Context/WebUser.php (hunk 2 above)
Comment (pierallard) [inline features/Context/WebUser.php]: :+1:

**Comment by pierallard** [comment]
GTM after '20' param fix

**Pull request merged**